### PR TITLE
Separate UID2 and EUID EIF build steps

### PR DESCRIPTION
**Pull request by cYKatherine**
Successful run: https://github.com/IABTechLab/uid2-operator/actions/runs/8428783502

### CODE REVIEW

File: .github/workflows/publish-aws-nitro-enclave-docker.yaml [hunk 1]
```diff
@@ -112,57 +112,134 @@ jobs:
...
+        uses: thetradedesk/git-restore-mtime-action@v1.3
+
+      - name: Free up space - delete preinstalled tools
+        run: |
+          rm -rf /opt/hostedtoolcache
 
       - name: Build UID2 AWS EIF
         id: build_uid2_eif
-        uses: IABTechLab/uid2-operator/.github/actions/build_aws_eif@v2
+        uses: IABTechLab/uid2-operator/.github/actions/build_aws_eif@main
```
Comment (cody-constine-ttd) [inline .github/workflows/publish-aws-nitro-enclave-docker.yaml:149]: We should probably stay with a specific version here.

File: .github/workflows/publish-aws-nitro-enclave-docker.yaml [hunk 2]
```diff
@@ -112,57 +112,134 @@ jobs:
...
         with:
-          name: aws-uid2-deployment-files-${{ steps.version.outputs.new_version }}
+          name: aws-uid2-deployment-files-${{ needs.start.outputs.new_version }}
           path: ${{ env.ARTIFACTS_BASE_OUTPUT_DIR }}/uid2
           if-no-files-found: error
-  
+    outputs:
+      uid2_enclave_id: ${{ steps.build_uid2_eif.outputs.enclave_id }}
+
+  buildEUIDEIF:
```
Comment (cody-constine-ttd) [inline .github/workflows/publish-aws-nitro-enclave-docker.yaml:163]: Could we combine this with the above and either just produce both or have an option in a drop down to pick

Comment (cYKatherine) [reply .github/workflows/publish-aws-nitro-enclave-docker.yaml:163]: I think the idea originally was to always generate an EIF for both UID2 and EUID.

File: .github/workflows/publish-aws-nitro-enclave-docker.yaml (hunk 1 above)
Comment (cYKatherine) [reply .github/workflows/publish-aws-nitro-enclave-docker.yaml:149]: `v2` was wrong. For operator we don't have versioning, we have to specify `main` branch. 

Alternatively, we can use `latest`, but the downside is that if we made some changes and didn't do a release, we won't pick up that change for the pipeline. That's why I kept it as `main`.

Comment (thomasm-ttd) [reply .github/workflows/publish-aws-nitro-enclave-docker.yaml:149]: main is the correct reference to use I think

**Review by thomasm-ttd — APPROVED** [review]

**Pull request merged**